lib/IDE/CodeCompletion.cpp

```diff
@@ -2168,6 +2168,12 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
     addValueBaseName(Builder, Name);
     setClangDeclKeywords(VD, Pairs, Builder);
 
+    // "not recommended" in its own getter.
```

Comment:
Will this impact keypaths?  I think that would still be "okay".  Technically you can also call the setter safely from the getter if it doesn't mutually recurse, but I don't care much about that case.

Reply:
It did. I updated the PR to ensure it doesn't mark it "NotRecommended" in case of keypaths or explicit self.
I want to mark it because the compiler emit warning for them:
```swift
class MyClass {
  var value: Int {
    get {
      value = 12 // warning: attempting to access 'value' within its own getter
                 // note: access 'self' explicitly to silence this warning
      return 1
    }
    set {
    }
  }
}
```

Reply:
SGTM, thanks!